unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Transform/DCLTransform.cs

```diff
@@ -67,5 +70,61 @@ public void UpdateFromModel(BaseModel model)
         public BaseModel GetModel() => DCLTransform.model;
         public int GetClassId() => (int) CLASS_ID_COMPONENT.TRANSFORM;
         public void UpdateOutOfBoundariesState(bool enable) { }
+        
+        public static void CapTransformGlobalValuesToMax(Transform transform)
+        {
+            bool positionOutsideBoundaries = transform.position.sqrMagnitude > MAX_TRANSFORM_VALUE * MAX_TRANSFORM_VALUE;
+            bool scaleOutsideBoundaries = transform.lossyScale.sqrMagnitude > MAX_TRANSFORM_VALUE * MAX_TRANSFORM_VALUE;
+            
+            if (positionOutsideBoundaries || scaleOutsideBoundaries)
+            {
+                Vector3 newPosition = transform.position;
+                if (positionOutsideBoundaries)
+                {
+                    if (Mathf.Abs(newPosition.x) > MAX_TRANSFORM_VALUE)
+                        newPosition.x = MAX_TRANSFORM_VALUE * Mathf.Sign(newPosition.x);
+
+                    if (Mathf.Abs(newPosition.y) > MAX_TRANSFORM_VALUE)
+                        newPosition.y = MAX_TRANSFORM_VALUE * Mathf.Sign(newPosition.y);
+
+                    if (Mathf.Abs(newPosition.z) > MAX_TRANSFORM_VALUE)
+                        newPosition.z = MAX_TRANSFORM_VALUE * Mathf.Sign(newPosition.z);
+                }
+                
+                Vector3 newScale = transform.lossyScale;
+                if (scaleOutsideBoundaries)
+                {
+                    if (Mathf.Abs(newScale.x) > MAX_TRANSFORM_VALUE)
+                        newScale.x = MAX_TRANSFORM_VALUE * Mathf.Sign(newScale.x);
+
+                    if (Mathf.Abs(newScale.y) > MAX_TRANSFORM_VALUE)
+                        newScale.y = MAX_TRANSFORM_VALUE * Mathf.Sign(newScale.y);
+
+                    if (Mathf.Abs(newScale.z) > MAX_TRANSFORM_VALUE)
+                        newScale.z = MAX_TRANSFORM_VALUE * Mathf.Sign(newScale.z);
+                }
+                
+                SetTransformGlobalValues(transform, newPosition, transform.rotation, newScale, scaleOutsideBoundaries);
+            }
+        }
+        
+        public static void SetTransformGlobalValues(Transform transform, Vector3 newPos, Quaternion newRot, Vector3 newScale, bool setScale = true)
```

Comment:
is it necessary to be static? could be an instance method? in the other hand it could be an extension method so the call is cleaner

Reply:
I moved that one into Utils as well, since it's needed by `CapGlobalValuesToMax`